src/controllers/legacycontrollersettings.h

```diff
@@ -423,6 +431,141 @@ class LegacyControllerEnumSetting
     friend class ControllerS4MK3SettingTest_ensureLibrarySettingValueAndEnumEquals;
 };
 
+class LegacyControllerColorSetting
+        : public LegacyControllerSettingFactory<LegacyControllerColorSetting>,
+          public AbstractLegacyControllerSetting {
+  public:
+    LegacyControllerColorSetting(const QDomElement& element);
+
+    virtual ~LegacyControllerColorSetting() = default;
```

Comment:
consider putting the `= default;` into the implementation. Since there call needs to be virtual, there is no advantage to it being in the header (because it can't be inlined).
Also use `override` when overriding instead of something that looks like a redeclaration.
```suggestion
    ~LegacyControllerColorSetting() override;
```